src/common/com/intellij/plugins/haxe/ide/annotator/HaxeSemanticAnnotator.java

```diff
@@ -0,0 +1,302 @@
+        HaxeSemanticError.addError(
+          element,
+          new HaxeSemanticError(
+            "Overriding nothing",
+            new HaxeSemanticIntentionAction("Fix override") {
+              @Override
+              public void run() {
+                HaxePsiUtils.replaceElementWithText(overrideAttribute, "");
+              }
+            }
+          )
+        );
+      } else if (overrideAttribute == null && requiredOverride) {
+        HaxeSemanticError.addError(
+          element,
+          new HaxeSemanticError(
+            "Must override"
+          )
+        );
+      }
+      //PackageChecker.check((HaxePackageStatement)element);
+    }
+
+    for (ASTNode node : element.getNode().getChildren(null)) {
+      analyze(node.getPsi());
+    }
+  }
+}
+
+class PackageChecker {
+  static public void check(final HaxePackageStatement element) {
+    final HaxeReferenceExpression expression = ((HaxePackageStatement)element).getReferenceExpression();
+    String packageName = (expression != null) ? expression.getText() : "";
+    PsiDirectory fileDirectory = element.getContainingFile().getParent();
+    List<PsiFileSystemItem> fileRange = PsiFileUtils.getRange(PsiFileUtils.findRoot(fileDirectory), fileDirectory);
+    fileRange.remove(0);
+    String actualPath = PsiFileUtils.getListPath(fileRange);
+    final String actualPackage = actualPath.replace('/', '.');
+    final String actualPackage2 = HaxeResolveUtil.getPackageName(element.getContainingFile());
+    // @TODO: Should use HaxeResolveUtil
+
+    for (String s : StringUtils.split(packageName, '.')) {
+      if (!s.substring(0, 1).toLowerCase().equals(s.substring(0, 1))) {
+        HaxeSemanticError.addError(element, new HaxeSemanticError("Package name '" + s + "' must start with a lower case character"));
+      }
+    }
+
+    if (!packageName.equals(actualPackage)) {
+      HaxeSemanticError.addError(
+        element,
+        new HaxeSemanticError(
+          "Invalid package name! '" + packageName + "' should be '" + actualPackage + "'",
+          new HaxeSemanticIntentionAction("Fix package") {
+            @Override
+            public void run() {
+              Document document =
+                PsiDocumentManager.getInstance(element.getProject()).getDocument(element.getContainingFile());
+
+              if (expression != null) {
+                TextRange range = expression.getTextRange();
+                document.replaceString(range.getStartOffset(), range.getEndOffset(), actualPackage);
+              }
+              else {
+                int offset =
+                  element.getNode().findChildByType(HaxeTokenTypes.OSEMI).getTextRange().getStartOffset();
+                document.replaceString(offset, offset, actualPackage);
+              }
+            }
+          }
+        )
+      );
+    }
+  }
+}
+
+class PsiFileUtils {
+
+  static public String getListPath(List<PsiFileSystemItem> range) {
```

Comment:
You are right. I was just experimenting, placing some yet-uncomplete code and using it to get the feeling. And this is the case and there are other cases. But at least it work in some simple cases, so it can be improved later.